frontend/src/features/recommendation/SearchFilter.tsx

```diff
@@ -25,29 +26,42 @@ function SearchFilter() {
     setExpanded(!expanded);
   };
 
+  const isFilterActive = () => {
+    for (const pair of filterParams.entries()) {
+      if (pair[0] in recommendationFilters && parseInt(pair[1], 10) !== 50) {
+        return true;
+      }
+    }
+    return false;
+  };
+
   return (
     <Box color="text.secondary">
-      <CollapseButton expanded={expanded} onClick={handleExpandClick} />
+      <Badge color="secondary" variant="dot" invisible={!isFilterActive()}>
+        <CollapseButton expanded={expanded} onClick={handleExpandClick} />
+      </Badge>
```

Comment:
Minor: it would be a bit more reusable to implement this badge inside `CollapseButton`,  
and to use it with something like `<CollapseButton withBadge={!isFilterActive()} />`.